cpp/src/arrow/array/array_dict.h

```diff
@@ -111,6 +111,10 @@ class ARROW_EXPORT DictionaryArray : public Array {
 
   const DictionaryType* dict_type() const { return dict_type_; }
 
+  bool IsNull(int64_t i) const {
+    return indices_->IsNull(i) || dictionary_->IsNull(GetValueIndex(i));
```

Comment:
Could you use `dictionary()` instead of touching `dictionary_` directory?

```suggestion
    return indices_->IsNull(i) || dictionary()->IsNull(GetValueIndex(i));
```

`dictionary_` is initialized lazily.

Reply:
Sure